tests/mir-opt/simplify_static_switch.rs

```diff
@@ -0,0 +1,29 @@
+// unit-test: SimplifyStaticSwitch
```

Comment:
Could you add more tests?
With all the extra checks I made you add + weird cfgs if possible.
See https://github.com/rust-lang/rust/pull/107009 for a few examples.

Reply:
I have added a few more tests, including checks for mutation/borrowing.